samples/sdl_gamecontroller/main.c

```diff
@@ -0,0 +1,48 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+int main() {
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER);
+
+  SDL_GameController *gameController = SDL_GameControllerOpen(0);
+
+  debugPrint("Num joysticks      : %02i\n", SDL_NumJoysticks());
+  debugPrint("GameController name: %s\n", SDL_GameControllerName(gameController));
+  debugPrint("Is GameController  : %02i\n", SDL_IsGameController(0));
+  debugPrint("GameController Map : %s\n", SDL_GameControllerMapping(gameController));
+  debugPrint("GameControllerEvent: %02i\n", SDL_GameControllerEventState(SDL_QUERY));
+
+  debugPrint("\n");
+  assert(SDL_IsGameController(0));
+
+  while (true) {
+    SDL_Event event;
+    while (SDL_PollEvent(&event)) {
+      if (event.type == SDL_CONTROLLERBUTTONDOWN) {
```

Comment:
nit: I assume there's also a `BUTTONUP`? That would be equally interesting.

Overall I still think that the state-based API is *much, much, much* more important than the event based API. I have never heard of anyone using the event based APIs for SDL2 in practical use-cases.

Reply:
I agree, however the spaminess without some kind of frame limiter would be immense and I don't think we should extend the samples that wide.

Reply:
You should at least add a comment then to make people aware of this option.

Also buttons can't be pressed that fast, so it won't spam a lot.
Meanwhile, any axis might jitter will probably cause > 100 of messages per second (which will drain any other message).

Reply:
Problem is that the non-event kind doesn't do "This has recently been pressed" but "is it currently pressed?". But your suggestion of adding an `XSleep()` could fix that problem.

Reply:
I'm not sure what you mean.

> Problem is that the non-event kind doesn't do "This has recently been pressed" but "is it currently pressed?".

This is the event-sample; anything related to the non-event sample is irrelevant for now.

*I don't see the issue of not knowing "This has recently been pressed" either; games are written as interactive applications, which draw on their own. There typically isn't any lockstep either as it would complicate the game logic. It's typically compensated by running multiple iterations of the physics simulation, which can benefit from updated inputs. You'd only need to know about higher sub-frame precision if your framerate is very low. But you wouldn't optimize your game for that as the player would be controlling "blindly".*

> But your suggestion of adding an XSleep() could fix that problem.

I did not recommend this, and it likely won't prevent spam as the events will still queue up probably.

Reply:
> Overall I still think that the state-based API is much, much, much more important than the event based API. I have never heard of anyone using the event based APIs for SDL2 in practical use-cases.

This was the part I was responding to (re spaminess), sorry for being vague.